src/marqo/core/inference/inference_models/hugging_face_model_properties.py

```diff
@@ -0,0 +1,119 @@
+import json
+from enum import Enum
+from json import JSONDecodeError
+from typing import Optional
+
+from huggingface_hub import hf_hub_download
+from huggingface_hub.utils import HfHubHTTPError
+from pydantic import Field, validator, root_validator
+
+from marqo.base_model import MarqoBaseModel
+from marqo.s2_inference.configs import ModelCache
+from marqo.s2_inference.logger import get_logger
+from marqo.tensor_search.models.private_models import ModelLocation, ModelAuth
+
+logger = get_logger(__name__)
+
+
+class PoolingMethod(str, Enum):
+    Mean = "mean"
+    CLS = "cls"
+
+
+class HuggingFaceModelProperties(MarqoBaseModel):
```

Comment:
It's probably worth making a `ModelProperties` class and having this class subclass from it. We may need an `OpenClipModelProperties` in the future.